sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -732,6 +755,39 @@ case class Cast(child: Expression, dataType: DataType, timeZoneId: Option[String
      """.stripMargin
   }
 
+  private def writeStructToStringBuilder(
+      st: Seq[DataType],
+      row: String,
+      buffer: String,
+      ctx: CodegenContext): String = {
+    val structToStringCode = st.zipWithIndex.map { case (ft, i) =>
+      val fieldToStringCode = castToStringCode(ft, ctx)
+      val funcName = ctx.freshName("fieldToString")
+      ctx.addNewFunction(funcName,
```

Comment:
no need to create a function, it's called only once.

Reply:
Actually we can create functions, for data types that appeared more than once among struct fields.

Reply:
BTW here we may hit 64kb compile error if there are a lot of fields in this struct. We should use `ctx.splitExpressionsWithCurrentInput`

Reply:
ok, l'll update soon